### ISSUE TITLE
Allow top N for aggregate 

### ISSUE THREAD

**Issue by danialfarid** [RediSearch/RediSearch#1035]
Following this question: https://stackoverflow.com/questions/59530799/redisearch-aggregate-return-top-5-of-each-group

It would be great to have the ability to return only the top N number of result just like `FIRST_VALUE` but with `TOP_N`.
I need to return more than one value from the sorted items in the group and currently there is only `TOLIST` which will return them all and is not practical for me if the list is too long or `FIRST_VALUE` which returns only one.

**Comment by mnunberg** [comment]
How long is too long? Perhaps we can provide a `LIST_TRIM` or similar which would operate on the output of `TOLIST`?

**Comment by danialfarid** [comment]
Yes I think that would work if the internal processing/time cost of FIRST_VALUE and LTRIM 1 would be about the same. But since we have FIRST_VALUE it seems natural to have FIRST_N_VALUES too

**Comment by jon-mercer** [comment]
I recently posted a [question on the Redis forum](https://forum.redis.com/t/ft-aggregate-sortby-groupby-tolist-doesnt-preserve-order-or-allow-me-to-truncate-the-list/2484) - now that I see this issue, I think a `FIRST_N_VALUES` reducer is exactly what I need.  To summarize, I would like the ability to have the output of `TOLIST` sorted and only return the first N results (for my use case, somewhere around 10 results).  

Without the ability to limit the output of results, each group has thousands of unsorted items in it, and I have to sort / truncate them to get the result I'm looking for.

So, consider this an emphatic +1 on this issue.
